hnn_core/network.py

```diff
@@ -205,10 +215,58 @@ def __init__(self, params):
         self.n_cells = sum(len(self.pos_dict[src]) for src in
                            self.cellname_list)
 
-        # The legacy code in HNN-GUI _always_ defines 2 'common' and 5
-        # 'unique' feeds. They are added here for backwards compatibility
-        # until a new handling of external NetworkDrives's is completed.
-        self._add_external_feeds()
+        if initialise_hnn_drives:
+            drive_specs = self._extract_drive_specs_from_hnn_params()
+            bias_specs = self._extract_bias_specs_from_hnn_params()
+
+            for drive_name in sorted(drive_specs.keys()):  # order matters
+                specs = drive_specs[drive_name]
+                if specs['type'] == 'evoked':
+                    self.add_evoked_drive(
+                        drive_name, specs['dynamics']['mu'],
+                        specs['dynamics']['sigma'],
+                        specs['dynamics']['numspikes'],
+                        specs['weights_ampa'], specs['weights_nmda'],
+                        specs['location'], seedcore=specs['seedcore'],
+                        space_constant=specs['space_constant'])
```

Comment:
syntax looks good to me. Maybe use keyword arguments? It helps avoid issues when passing arguments

Reply:
You mean passing with '**'?

Reply:
no, I mean, instead of doing:

```py
y = exponent(2, 3)
```

do

```py
y = exponent(base=2, power=3)
```

Reply:
actually here I think you could also use `**specs['dynamics']` after passing the rest of the arguments like I mention above